internal/repository/repository.go

```diff
@@ -580,16 +580,22 @@ func (r *Repository) SetIndex(i restic.MasterIndex) error {
 	return r.prepareCache()
 }
 
-// LoadIndex loads all index files from the backend in parallel and stores them
-// in the master index. The first error that occurred is returned.
-func (r *Repository) LoadIndex(ctx context.Context) error {
+func (r *Repository) LoadIndex(ctx context.Context, p *progress.Counter) error {
 	debug.Log("Loading index")
 
+	var numIndexFiles int
+	r.List(ctx, restic.IndexFile, func(id restic.ID, size int64) error {
```

Comment:
It would be good to use `be, err := backend.MemorizeList(ctx, repo.Backend(), restic.IndexFile)` to avoid listing the index files twice.

Reply:
I am unfamiliar with the function of MemorizeList. How should I use it to track how many index files need to be loaded?

Reply:
Hmm, using that function is trickier than I've thought. I'll have a look once the other review comments are resolved.

Reply:
Thanks for the feedback, I have resolved the other issues. 

BTW, when I was running the tests I noticed that there were some failed test cases. To test if theses were the result of my own changes, I ran the test cases on the main branch and noticed that the same test cases failed, so I'm assuming this expected behavior. Could you explain why this is the case?

Reply:
As the test failures are caused by `Error: cmd/restic/cmd_cat.go:175:24: not enough arguments in call to repo.LoadIndex` they are definitely a result of your changes. For me the `master` branch both compiles and successfully runs the tests, see for example https://github.com/restic/restic/actions/runs/5848699634 .

Reply:
So, that was quite tricky:

This PR was based on a version of the `master` branch that did not yet include #4334, which also modified `cmd_cat.go`. The CI tests run for a commit that is a merge of a PR and the current master branch. Consequently, the new code from #4334 was missing the modifications to its `LoadIndex(...)` call.

Although this behavior may seem strange, it's actually a good idea. The CI tests should not only answer the question whether the tests run successfully for the code in the PR, but rather whether it will work once merged!

The fix for that was to rebase the PR onto the current master branch.

The next part of the test failures were caused by listing the index twice. The integration tests verify that we don't introduce additional `List()` calls to the backend. I've fixed that by using `MemorizeList`.